tests/helpers/snapshot.js

```diff
@@ -0,0 +1,58 @@
+import http from 'http'
+import puppeteer from 'puppeteer'
+
+async function snapshot(page, name, options = {}) {
+  await page.addScriptTag({
+    url: 'http://localhost:5338/percy-agent.js'
+  })
+
+  let dom = await page.evaluate(options => {
+    return new PercyAgent({ handleAgentCommunication: false })
+      .domSnapshot(document, options)
+  }, options)
+
+  let data = JSON.stringify({
+    name,
+    url: page.url(),
+    domSnapshot: dom,
+    clientInfo: 'acceptance-micro-sdk',
+    ...options
+  })
+
+  let req = http.request({
+    port: 5338,
+    method: 'POST',
+    path: '/percy/snapshot',
+    headers: {
+      'Content-Type': 'application/json',
+      'Content-Length': data.length
+    }
+  })
+
+  req.write(data)
+  req.end()
+
+  return dom
+}
+
+export default async function launch(callback) {
+  let browser = await puppeteer.launch({
+    headless: true,
+    ignoreHTTPSErrors: true,
+    args: [
+      '--disable-gpu',
+      '--no-sandbox',
+      '--single-process',
+      '--disable-dev-profile'
+    ]
+  })
+
+  try {
+    let page = await browser.newPage()
+    await page.setBypassCSP(true)
```

Comment:
Are we sure we want to set this here? Are there cases where we'll want CSP bypassing off?

Reply:
Without it, I can't even inject the client script into this local express server. I figured if it's used in PercyScript, probably should just use it here in this mini-clone.

Reply:
Ah, gotcha. Makes sense